K59/Nguyen-Van-Sang/computer_network/VLAN, VXLAN.md

```diff
@@ -0,0 +1,281 @@
+
+Hình 2.4 Cơ chế khám phá và học địa chỉ trong VXLAN
+
+![hinh2.4](imageNet/vxlan04.png)
+
+**Bước 1**: Host A gửi 1 gói tin ARP request với IP đích(IP-B) để yêu cầu MAC đích(MAC-B) tới VTEP-1
+
+**Bước 2**: VTEP-1 nhận được ARP request, kiểm tra và thấy chưa có ánh xạ cho IP-B. Nó đóng gói ARP request thành 1 gói IP multicast. Gói tin VXLAN này có IP nguồn là IP của VTEP-1 (IP-1), IP đích là IP của nhóm Multicast (239.1.1.1), và được chuyển tới tất cả các VTEP khác trong nhóm Multicast 239.1.1.1 
+
+**Bước 3**: Tất cả các VTEPs trong nhóm Multicast 239.1.1.1 (gồm VTEP-2, VTEP-3) nhận được gói tin multicast này. Các VTEPs này kiểm tra VNID trong tiêu đề VXLAN của gói tin, nếu VNID 10 này được cấu hình thì các VTEPs này sẽ thêm ánh xạ MAC-to-VTEP vào local table (bảng cục bộ) của nó. Local table lúc này có dạng:
+
+
+MAC Address | VXLAN ID | Remote VTEP
+---------|----------|---------
+ MAC-A | 10 | IP-1
+
+**Bước 4**: Các VTEPs, bỏ phần tiêu đề ngoài rồi gửi gói tin gốc ARP request (chứa IP đích B) cho các host liên kết với nó. Các host so sánh IP của mình với IP đích, nếu không trùng thì gói tin bị hủy bỏ. Host B nhận được ARP request, kiểm tra trùng với IP đích nó tiến hành đóng 1 gói ARP response chứa MAC-B gửi phản hồi tới VTEP-2, đồng thời nó cũng học ánh xạ IP-A-to-MAC-A
+
+**Bước 5**: VTEP-2 nhận được gói ARP response với MAC đích là MAC-A. Lúc này nó kiểm tra local table của nó và biết là ánh xạ MAC-A-to-IP-1 đã được thiết lập. VTEP-2 đóng gói ARP response này thành 1 gói VXLAN unicast rồi chuyển tới VTEP-1. Gói unicast này có IP nguồn là IP-2, IP đích là IP-1.
+
+**Bước 6**: VTEP-1 nhận được gói unicast từ VTEP-2. Nó de-encapsulation và chuyển ARP response đến Host A, đồng thời nó cũng lập 1 ánh xạ MAC-to-VTEP vào local table của mình với những thông tin sau:
+
+MAC Address | VXLAN ID | Remote VTEP
+---------|----------|---------
+ MAC-B | 10 | IP-2
+
+ **Bước 7**: Bây giờ việc học và ánh xạ địa chỉ giữa Host A và Host B, VTEP-1 và VTEP-2 hoàn thành, đường hầm VXLAN giữa chúng được thiết lập, các gói tin Unicast tiếp theo giữa Host A và Host B sẽ sử dụng đường hầm này để giao tiếp
+
+## 5. Chuyển tiếp gói tin trong VXLAN
+
+Các quy tắc chuyển tiếp của VTEP:
+
+1. Nếu MAC nguồn và MAC đích đến đều nằm cùng 1 Server lưu trữ thì gói tin được chuyển tới Switch (có cài đặt VTEP), được switch chuyển đến máy đích luôn và không có VXLAN encapsulation/de-encapsulation được thực hiện
+2. Nếu MAC đích không hoạt động cùng Server chứa host gửi gói tin, thì gói tin được đóng thành gói VXLAN bởi VTEP nguồn và được gửi tới VTEP tiếp theo dựa trên local table của nó. Tới VTEP có kết nối với host đích gói tin sẽ được loại bỏ các phần tiêu đề ngoài, và gói tin gốc sẽ được gửi tới host đích
+3. Với các gói tin Unicast không rõ, hoặc multicast. VTEP sẽ đóng gói thành gói VXLAN multicast rồi gửi đến các VTEP khác thuộc cùng nhóm multicast của 1 VXLAN segment (lúc này Outer Dest IP là IP của nhóm multicast)
+
```

Comment:
Tài liệu : https://www.arista.com/assets/data/pdf/Whitepapers/Arista_Networks_VXLAN_White_Paper.pdf

Trang 3, mục VXLAN frame encapsulation and forwarding